v2/pkg/engine/resolve/loader.go

```diff
@@ -936,11 +936,37 @@ func (l *Loader) optionallyOmitErrorFields(values []*astjson.Value) {
 
 // optionallyOmitErrorLocations removes the "locations" object from all values.
 func (l *Loader) optionallyOmitErrorLocations(values []*astjson.Value) {
-	if !l.omitSubgraphErrorLocations {
-		return
-	}
 	for _, value := range values {
-		if value.Exists("locations") {
+		if !value.Exists("locations") {
+			continue
+		}
+
+		// If the flag is set, delete all locations
+		if l.omitSubgraphErrorLocations {
+			value.Del("locations")
+			continue
+		}
```

Comment:
I think value.Del is a no-op if it doesn't exist. You could add it to the other if statement 

```go
if !value.Exists("locations") || l.omitSubgraphErrorLocations {
  value.Del("locations")	
  continue
}
```